src/tribol/mesh/MfemData.hpp

```diff
@@ -1572,11 +1621,21 @@ class MfemJacobianData {
    */
   mfem::Array<int> block_offsets_;
 
+  /**
+   * @brief Array of offsets equal to number of displacement degrees of freedom
+   */
+  mfem::Array<int> disp_offsets_;
+
   /**
    * @brief List giving global parent vdof given the submesh vdof
    */
   mfem::Array<int> submesh2parent_vdof_list_;
 
+  /**
+   * @brief Submesh to parent transfer operator
+   */
+  std::unique_ptr<mfem::HypreParMatrix> submesh_parent_vdof_xfer_;
```

Comment:
Is this being introduced as a refactor or associated with an additional capability?

Reply:
Additional capability; future refactor depending on how well it works with problems.